.github/workflows/kitchen.yml

```diff
@@ -16,27 +16,87 @@ defaults:
-      KITCHEN_LOCAL_YAML: kitchen.exec.windows.yml
+      CHEF_LICENSE_KEY: free-79df705d-b685-419a-8b68-88401f74ff72-3999
+      HAB_ORIGIN: gha
+      HAB_BLDR_CHANNEL: base-2025
+      HAB_REFRESH_CHANNEL: base-2025
+      HAB_AUTH_TOKEN: ${{ secrets.HAB_AUTH_TOKEN }}
     steps:
-    - uses: actions/checkout@v5
-    - name: Install Chef
-      uses: actionshub/chef-install@main
-      with:
-        version: "25.5.1084"
-    - name: Kitchen Test
-      run: kitchen verify ${{ matrix.os }}
+    - uses: actions/checkout@v4
+    - name: Install Habitat CLI
+      run: |
+        # Install Habitat using Chocolatey
+        choco install -y habitat
+        refreshenv
+        # Ensure hab is in PATH
+        $env:PATH = "C:\Program Files\Habitat;" + $env:PATH
+        echo "PATH=$env:PATH" | Out-File -FilePath $env:GITHUB_ENV -Append -Encoding utf8
+        hab license accept
+        hab --version
+    - name: Generate Habitat Origin Key
+      run: |
+        hab origin key generate "$env:HAB_ORIGIN"
+    - name: Install Chef CLI
+      run: |
+        hab pkg install --binlink --force chef/chef-cli --channel unstable
+    - name: Build Habitat Package
+      run: |
+        # Move to repo root
+        cd ..
+        hab pkg build .
+
+        # Find the built package
+        $pkg_path = Get-ChildItem -Path "results" -Filter "$env:HAB_ORIGIN-chef-infra-client-*.hart" |
+                    Sort-Object LastWriteTime -Descending |
+                    Select-Object -First 1
+
+        if ($pkg_path) {
+            Write-Host "Built package: $($pkg_path.Name)"
+            Copy-Item $pkg_path.FullName -Destination "kitchen-tests\chef-infra-client-latest.hart"
+        } else {
+            Write-Error "No package built"
+            Exit 1
+        }
+    - name: Install Chef Infra Client Habitat Package
+      run: |
+        hab pkg install chef-infra-client-latest.hart
+        hab pkg exec $env:HAB_ORIGIN/chef-infra-client chef-client --version
+    - name: Run end-to-end recipes in Local Mode
+      run: |
+        # Install the Policyfile (resolves dependencies)
+        hab pkg exec chef/chef-cli chef-cli install Policyfile.rb
+        if ($LASTEXITCODE -ne 0) {
+            Write-Error "Failed to install Policyfile"
+            Exit 1
+        }
+        # Export the policy and all cookbooks to a sandbox directory
+        # This is exactly what Test Kitchen does
+        hab pkg exec chef/chef-cli chef-cli export Policyfile.lock.json policy-export --force
+        if ($LASTEXITCODE -ne 0) {
+            Write-Error "Failed to export Policyfile"
+            Exit 1
+        }
+        # Run chef-client in local mode from the exported directory
+        # Just like Test Kitchen does: cd <export-dir> && chef-client -z
+        cd policy-export
+        hab pkg exec $env:HAB_ORIGIN/chef-infra-client chef-client -z
+        if ($LASTEXITCODE -ne 0) {
+            Write-Error "Chef Infra Client run failed"
+            Exit 1
+        }
```

Comment:
This is... doing what kitchen does... but outside of kitchen?

Why? This feels super hacky.

AND it means no one can run tests locally anymore.

I know there's pressure to get Chef 19 out the door, but we need to actually do the work to make this work properly.

Reply:
@jaymzh @Stromweld @tpowell-progress   test-kitchen-enterprise does not yet have windows support. Instead of waiting for that to be out and not have recipe testing of chef19 hab package at all, this way we will at least start getting end-to-end recipe testing for windows habitat package of infra client 19. I know this looks a loot right now but without recipe testing being done before GA, we are waiting lot of issues getting unnoticed and pilling up in the end for us.

Reply:
Biggest issue with both community and CTKE is that neither supports new pathing for habitat pkg based chef-client. It's hard coded to pass "/opt/chef/bin/chef-client" and a bunch of flags. The provisioner needs to be updated to fix that then CTKE with our updated provisioner will work.